File: frontend/src/api/summary.ts
export interface UserSummary {
  userId: number;
  unique_word_count: number;
  unique_words: string[];
}

export interface SummaryResponse {
  top_users: UserSummary[];
  common_words: string[];
}

export async function fetchSummary(): Promise<SummaryResponse> {
  const res = await fetch('/summary');
  if (!res.ok) throw new Error('Failed to fetch summary');
  return res.json();
}

File: backend/app/models/summary.py
from pydantic import BaseModel
from typing import List, Dict

class UserSummary(BaseModel):
    userId: int
    unique_word_count: int
    unique_words: List[str]

class SummaryResponse(BaseModel):
    top_users: List[UserSummary]
    common_words: List[str]
